### DISCUSSION TITLE
reading xbt edf file

### DISCUSSION THREAD

**Discussion by ecamossi** [dankelley/oce#2257] [Q&A] [answered]
Hello,

I'm trying to load some xbt  data exported to MK21 EXPORT DATA FILE  (EDF) using `read.xbt.edf,` , but I'm getting the following error:

`programming error: increase #lines read for header`

After some test comparing my data with the sample data, I think I figured out that the issue is due to the number of variable data included  in the export, in this case 5

`// Data Fields`
`Num Data Fields  :  5`
`Field1           :  Time (sec)`
`Field2           :  Resistance (ohms)`
`Field3           :  Depth (m)`
`Field4           :  Temperature (°C)`
`Field5           :  Sound Velocity (m/s)`

In particular, removing the first two columns seems to make the data as expected by the reader. 

Is there any suggestion on how to fix this issue? I can of course read the text file but I would like to take advantage of the nice features that `oce` provides by default for the data types supported.

Thanks for any hints!
-ec

**Comment by dankelley** [comment]
Hi, and thanks for the report.  Is there a chance you could email the data file to me (kelley.dan at gmail dot com) and Clark (clark.richards at the same) so we can look?

If the data are non-private, another alternative is to make a copy, chop out all but the first few data lines (that is, retaining the header), rename to have suffix .txt, and attach to the issue.

Also, can you build oce from source?  We don't make releases to CRAN frequently.  But, even if not, I can show you how to get the fix to work.

**Comment by ecamossi** [comment]
Hello, Thanks for replying so quickly!Data are indeed private but tomorrow I will prepare a mock up to share as close as possible to the real file. I haven’t tried to build the package, I just installed if from CRAN, but if you can guide me a bit I think I can do it.I will check also if we have a manual for the instrument, I’m not sure about how the export has been generated but it’s likely it comes directly from the instrument itself. Thanks a lot! Kind regards, - ElenaOn 11 Nov 2024, at 18:39, Dan Kelley ***@***.***> wrote:﻿
Hi, and thanks for the report.  Is there a chance you could email the data file to me (kelley.dan at gmail dot com) and Clark (clark.richards at the same) so we can look?
If the data are non-private, another alternative is to make a copy, chop out all but the first few data lines (that is, retaining the header), rename to have suffix .txt, and attach to the issue.
Also, can you build oce from source?  We don't make releases to CRAN frequently.  But, even if not, I can show you how to get the fix to work.

**Comment by dankelley** [comment, marked as answer]
What I really need is the data-header line, i.e. the last line before the actual data. I need to devise a regular-expression to identify that.  Whether other things work is a question -- if the format is not like the files I already have, then likely very little of the metadata will be recovered without recoding.

Please post that line on the issue page (https://github.com/dankelley/oce/issues/2258), not here in the discussion.  Issues are where I look for information.

**Comment by ecamossi** [comment]
The original question and issue has been answered in issue [#2258 ](https://github.com/dankelley/oce/issues/2258). Thanks!
